Jenkinsfile

```diff
@@ -1,59 +1,100 @@
-                '''
+            steps {
+                sh './gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
+                    -PartifactoryUrl=$ARTIFACTORY_URL \
+                    -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                    -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                    --gradle-user-home GRADLE_USER_HOME \
+                    --continue --info --stacktrace \
+                    clean assemble'
             }
         }
         stage('Test') {
-            mdk_builder.inside{
-                sh '''
-                export GRADLE_USER_HOME=$(pwd)/.gradle
-                xvfb-run ./gradlew -PbuildNumber=$BUILD_NUMBER -PbuildAccess=$BUILD_ACCESS -PbuildTag=$GIT_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW -PtestrailHost=$TESTRAIL_HOST -PtestrailUser=$TESTRAIL_CREDENTIALS_USR -PtestrailPassword=$TESTRAIL_CREDENTIALS_PSW -PtestrailSuiteId=$TESTRAIL_SUITE_ID -PadditionalTestArguments=$ADDITIONAL_TEST_ARGUMENTS --gradle-user-home GRADLE_USER_HOME --continue --info --stacktrace check testrailPublish
-                '''
+            steps {
+                sh 'xvfb-run ./gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
+                    -PartifactoryUrl=$ARTIFACTORY_URL \
+                    -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                    -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                    -PtestrailHost=$TESTRAIL_HOST \
+                    -PtestrailUser=$TESTRAIL_CREDENTIALS_USR \
+                    -PtestrailPassword=$TESTRAIL_CREDENTIALS_PSW \
+                    -PtestrailSuiteId=$TESTRAIL_SUITE_ID \
+                    -PadditionalTestArguments=$ADDITIONAL_TEST_ARGUMENTS \
+                    --gradle-user-home GRADLE_USER_HOME \
+                    --continue --info --stacktrace \
+                    check testrailPublish'
+            }
+        }
+        stage('Publish Snapshot') {
+            when { branch 'develop' }
+            steps {
+                sh './gradlew \
+                -PbuildNumber=1000 \
+                -PbuildAccess=$BUILD_ACCESS \
+                -PbuildTag=$TAG_NAME \
+                -PartifactoryUrl=$ARTIFACTORY_URL \
+                -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                -PartifactoryRepository=maven-libs-snapshot-local \
```

Comment:
I believe there are Jenkins credentials that include this string. We can use that instead of hard coding.